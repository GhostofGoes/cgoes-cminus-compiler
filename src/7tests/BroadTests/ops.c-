// just test all the ops mostly in a singlton expression
//
int y[10];
bool x;

main()
{
    int a;
    int b;
    int c;
    bool z;

    y[0] = 50;
    y[5] = 1;
    y[8] = 2;

    a = 50;
    b = 1;
    c = 8;

    x = false;
    z = true;

    outputb(a <= 1);		// F 
    outputb(b <= 50);		// T 
    outputb(50 <= 50);		// T 
    outnl();

    outputb(50<1);		// F 
    outputb(b<a);		// T 
    outputb(a<a);		// F 
    outnl();

    outputb(50>1);		// T 
    outputb(1>50);		// F 
    outputb(50>50);		// F 
    outnl();

    outputb(50 >= 1);		// T 
    outputb(1 >= 50);		// F 
    outputb(50 >= 50);		// T 
    outnl();

    outputb(y[0] == b);		// F 
    outputb(a == b);		// F 
    outputb(b == b);		// T 
    outnl();

    outputb(y[0] != a);		// F 
    outputb(y[0] != b);		// T 
    outputb(a != b);		// T 
    outputb(b != b);		// F 
    outnl();

    outputb(true  &  true);	// T 
    outputb(true  &  false);	// F 
    outputb(false  &  true);	// F 
    outputb(false  &  false);	// F 
    outnl();

    outputb(true  |  true);	// T 
    outputb(z  |  false);	// T 
    outputb(false  |  true);	// T 
    outputb(false  |  false);	// F 
    outnl();

    outputb( ! true);		// F 
    outputb( ! x);		// T 
    outnl();

    output(5 + 50);		// 55 
    output(y[c] + 50);		// 52 
    outnl();

    output(5 - 50);		// -45 
    output(30 - 5 - 50);	// -25 
    outnl();

    output(-a);			// -50 
    output(-0);			// 0
    output(-1);			// -1 
    outnl();

    output(2*3*5*7*11);		// 2310
    output(50*y[8]);		// 100 
    output(y[8]*50);		// 100 
    outnl();

    output(50/y[8]);		// 25 
    output(y[8]/50);		// 0 
    output(y[8]/b);		// 2 
    output(210/7/3);		// 10 
    outnl();

    output(211%7);              // 1 
    output(211%6);              // 1 
    output(211%5);              // 1 
    output(211%4);              // 3 
    output(211%100%3);          // 2 
    outnl();

    output(a = b = c = 43);     // 43
    output(a);                  // 43
    a = (b=3) * (c = 4);    
    output(a);                  // 12
    outnl();

    a = 44;
    b = 2;
    output(a+=b);                // 46
    output(a-=b);                // 44
    output(a-=a);                // 0
    output(a+=b+=10);            // 12
    outnl();

    output(3-4-5);               // -6
    output(211/17/5);            //  2 
    output(211/(17/5));            // 70
    outputb((true | false) & false); // F  
    outputb(true | false & false);   // T
    outputb(true | (false & false)); // T
    outnl();
}
