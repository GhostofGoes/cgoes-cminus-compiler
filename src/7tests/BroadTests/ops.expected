Number of warnings: 0
Number of errors: 0
------------------------------------

Memory Configuration: Data Addresses: 0-9999  Instruction Addresses: 0-9999
Loading file: ops.tm
F T T 
F T F 
T F F 
T F T 
F F T 
F T T F 
T F F F 
T T T F 
F T 
55 52 
-45 -25 
-50 0 -1 
2310 100 100 
25 0 2 10 
1 1 1 3 2 
43 43 12 
46 44 0 12 
-6 2 70 F T T 
Status: Halted
Bye.
